### DISCUSSION TITLE
Project Activity

### DISCUSSION THREAD

**Discussion by codetricity** [caitlynbakery/flutter_video_display_8#1] [Ideas]
# Objective - Verify video player works with THETA X video files from local storage

## Flutter Video Display (FVD) 1 Flutter video_player example

@jcasman and @caitlynbakery should provide feedback on direction of project

Run example

* https://github.com/flutter/plugins/tree/main/packages/video_player/video_player/example
* test with video file from camera that is manually transferred into the `assets/` folder of the project.

Focus on a smaller apps first:

## FVD Ep 2 - image download
just does image download and then saves the image to local mobile phone storage.

## FVD Ep 3 - video download

Download a video file from the THETA X and save to local app storage

## FVD Ep 4 - play video from local storage after download from camera

need to figure out how to play the video file when it is not in `assets/` folder

![image](https://user-images.githubusercontent.com/5799860/178025354-73bbacd1-6ecf-4dab-8b73-04d3903fd781.png)

---

## Status of Playing Video File Directly from Camera

The camera web server may not be designed to allow the videos to be played directly from the camera.  Developers may need to first download the video file, then play it.

I am unable to get the video from the THETA to play in a Flutter app when using the remote URL with the camera as the HTTP endpoint with these packages:

https://pub.dev/packages/video_player
https://pub.dev/packages/better_player

**Comment by jcasman** [comment]
Trying an improvement of the FVD Ep 4 programming flow chart. The main goal was to improve the yes/no loop.

<img width="788" alt="Screen Shot 2022-07-08 at 5 08 15 PM" src="https://user-images.githubusercontent.com/1075520/178083856-d8417a10-3d0d-4156-87bb-e3744bafed55.png">

**Reply by jcasman** [reply]
Used https://www.edrawmax.com/ to draw the diagram

**Reply by codetricity** [reply]
> Used https://www.edrawmax.com/ to draw the diagram

Are you using a commercial license for EdrawMax?  Is it free?

We're currently using the free version of Figma Design.  I am next going to try to use Figma FigJam.

Do you have any more insight into the conventions for diagrams?

The Figma Design drawings look more aesthetic IMO are closer to a standard design tool for mobile app design, but it's not optimized for workflow flowcharts or process flowcharts.  If we all use Figma, we can use some of the Figma collaboration features.

What are the benefits of using EdrawMax?

## Figma Benefits

* get more practice with a common design tool used by the industry
* features and concepts are close to [Sketch](https://www.sketch.com/)
    * main reason we are not using Sketch is because it is only available on the Mac
* It can import and export Sketch files
* compatible with design templates like Front

## Figma Cons

* kind of difficult to use
* not optimized for flowcharts

**Comment by codetricity** [comment]
# Flutter Video Display (FVD) Ep 1

## Title: Play RICOH THETA Videos on Mobile Phones

## Overview

Play local video files both in equirectangular and 360 navigation

## Teaching Objects

Students following our tutorials will learn:

* RICOH THETA video files are standard MP4 video files in a 2:1 aspect ratio and can be viewed with standard mobile app viewers
* 360 navigation will require a special 360 viewer

## Recommended Technology

You can deviate from the recommended technology if you need to

* [video_player](https://pub.dev/packages/video_player) from flutter.dev - 1736 likes and 100% popularity

Optional

* see if [video_360](https://pub.dev/packages/video_360) works from a local file - 7 likes, 74% popularity

Alternate

* [chewie](https://pub.dev/packages/chewie) - 1039 likes, 99% popularity

## Recommended App Strategy

https://github.com/flutter/plugins/tree/main/packages/video_player/video_player/example
test with video file from camera that is manually transferred into the assets/ folder of the project.

Getting a video from a network URL is optional.  However, it is desirable to try and put a RICOH 360 video up.  The example in the video_player example directory uses a file that is on a GitHub Pages site.  You can try putting a video file in the docs directory of the repo and make the docs directory a github page

Example

* https://codetricity.github.io/flutter_video_display/ageda.MP4
* https://codetricity.github.io/flutter_video_display/coffee.MP4

---

## Points for future tutorials and not covered in this tutorial

* location of video file on RICOH THETA camera is available with a URL that is accessible with a HTTP GET request
* the URL is available from stopCapture

**Reply by caitlynbakery** [reply]
Ok, I have the network and asset video working at the moment, but I'll continue working on it in the afternoon with the list view and 360.

**Reply by codetricity** [reply]
For the 360 video display, first try to get the video file from the network resource.  Example: https://codetricity.github.io/flutter_video_display/ageda.MP4

From looking at the source, it does not appear to me that it is possible to play the 360 video from local storage.  I opened up a GitHub issues on the flutter_video_360 repo.

https://github.com/kilroy80/flutter_video_360/issues/9

**Comment by caitlynbakery** [comment]
I added the ability to get the file from Firebase Storage. 
Link to the media file: https://firebasestorage.googleapis.com/v0/b/oppkeytheta.appspot.com/o/ageda.MP4?alt=media

**Reply by jcasman** [reply]
Do you have to add anything specific to handle a specific cloud storage service? Or if it has a URL that automatically plays the video it will just work?

**Reply by caitlynbakery** [reply]
To get the video to work you just have to replace the url with the media file url.

```dart
child: Video360View(
                onVideo360ViewCreated: _onVideo360ViewCreated,
                url:
https:firebasestorage.googleapis.com/v0/b/oppkeytheta.appspot.com/o/R0010926.MP4?alt=media',
```

**Reply by jcasman** [reply]
What does the ?alt=media tag at the end of the URL do?

**Reply by caitlynbakery** [reply]
It doesn't work consistently if the ?alt=media tag isn't at the end of the URL.

**Reply by codetricity** [reply]
> What does the ?alt=media tag at the end of the URL do?

I think it makes the file accessible as an image.  Otherwise, I don't think you can use it consistently.  I believe it's the format of the API endpoint for the image files on Firebase Storage.  As the API endpoint is not a conventional web server like you might be used to, the media may have the same format that you're used to seeing on a web server. 

The main paradigm to understand is that Firebase Storage is not a web server and thus the media may be accessed by a different format from a web server.  Firebase Storage is a storage system.
